gnome-initial-setup/pages/account/gis-account-page-eula-dialog.c

```diff
@@ -0,0 +1,319 @@
+      g_error_free (error);
+      g_free (path);
+
+      return NULL;
+    }
+
+  model = ev_document_model_new_with_document (document);
+  view = ev_view_new ();
+  ev_view_set_model (EV_VIEW (view), model);
+
+  g_object_unref (document);
+  g_object_unref (model);
+
+  gtk_widget_set_hexpand (view, TRUE);
+  gtk_widget_set_vexpand (view, TRUE);
+
+  return view;
+}
+
+static void
+evince_view_external_link_cb (EvView *ev_view,
+                              EvLinkAction *action,
+                              GisAccountPageEulaDialog *dialog)
+{
+  const gchar *uri;
+  GFile *file;
+  GtkWidget *view;
+
+  uri = ev_link_action_get_uri (action);
+  file = g_file_new_for_uri (uri);
+
+  if (g_file_has_uri_scheme (file, "file"))
+    {
+      view = load_evince_view_for_file (file);
+      if (view != NULL)
+        present_view_in_modal (dialog, view, _("Terms of Use"), TRUE);
+    }
+  else if (g_str_has_prefix (uri, LICENSE_SERVICE_URI))
+    {
+      view = load_license_service_view ();
+      present_view_in_modal (dialog, view, _("Third Party Software"), FALSE);
+    }
+
+  g_object_unref (file);
+}
+
+static void
+load_terms_view (GisAccountPageEulaDialog *dialog)
+{
+  GisAccountPageEulaDialogPrivate *priv = gis_account_page_eula_dialog_get_instance_private (dialog);
+  GFile *file;
+  GtkWidget *widget;
+
+  file = get_terms_document ();
+  if (file == NULL)
+    return;
+
+  priv->evince_view = load_evince_view_for_file (file);
+  g_object_unref (file);
+
+  if (priv->evince_view == NULL)
+    return;
+
+  widget = priv->scrolled_window;
+  gtk_container_add (GTK_CONTAINER (widget), priv->evince_view);
+  gtk_widget_show (priv->evince_view);
+
+  g_signal_connect (priv->evince_view, "external-link",
+                    G_CALLBACK (evince_view_external_link_cb), dialog);
+}
+
+static void
+load_css_overrides (GisAccountPageEulaDialog *dialog)
+{
+  GtkCssProvider *provider;
+  GError *error = NULL;
+
+  provider = gtk_css_provider_new ();
+  gtk_css_provider_load_from_resource (provider, "/org/gnome/initial-setup/endless-eula-page.css");
+
+  if (error != NULL)
+    {
+      g_warning ("Unable to load CSS overrides for the endless-eula page: %s",
+                 error->message);
+      g_error_free (error);
+    }
+  else
+    {
+      gtk_style_context_add_provider_for_screen (gtk_widget_get_screen (GTK_WIDGET (dialog)),
+                                                 GTK_STYLE_PROVIDER (provider),
+                                                 GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+    }
+
+  g_object_unref (provider);
+}
+
+static void
+gis_account_page_eula_dialog_constructed (GObject *object)
+{
+  GisAccountPageEulaDialog *dialog = GIS_ACCOUNT_PAGE_EULA_DIALOG (object);
+
+  G_OBJECT_CLASS (gis_account_page_eula_dialog_parent_class)->constructed (object);
+
+  load_css_overrides (dialog);
+  gtk_widget_show_all (GTK_WIDGET (dialog));
```

Comment:
The caller also calls `show_all()` on the dialog. Can probably remove one or the other